invenio_rdm_records/services/access/service.py

```diff
@@ -353,44 +359,64 @@ def _check_grant_subject(self, identity, grant):
             return False
 
     @unit_of_work()
-    def create_grant(self, identity, id_, data, expand=False, uow=None):
-        """Create an access grant for a record (resp. its parent)."""
+    def create_grants(self, identity, id_, data, expand=False, uow=None):
+        """Create access grants for a record (resp. its parent)."""
         record, parent = self.get_parent_and_record_or_draft(id_)
 
         # Permissions
         self.require_permission(identity, "manage", record=record)
 
         # Validation
-        data, __ = self.schema_grant.load(
+        data, __ = self.schema_grants.load(
             data, context={"identity": identity}, raise_errors=True
         )
 
-        for grant in parent.access.grants:
-            if (
-                grant.subject_id == data["subject"]["id"]
-                and grant.subject_type == data["subject"]["type"]
-            ):
-                raise GrantExistsError()
+        grants = data["grants"]
+
+        new_grants = []
+        for grant in grants:
+            for existing_grant in parent.access.grants:
+                if (
+                    existing_grant.subject_id == grant["subject"]["id"]
+                    and existing_grant.subject_type == grant["subject"]["type"]
+                ):
+                    raise GrantExistsError()
+
+        for grant in grants:
+            # Creation
+            new_grant = parent.access.grants.create(
+                subject_type=grant["subject"]["type"],
+                subject_id=grant["subject"]["id"],
+                permission=grant["permission"],
+                origin=grant.get("origin"),
+            )
 
-        # Creation
-        grant = parent.access.grants.create(
-            subject_type=data["subject"]["type"],
-            subject_id=data["subject"]["id"],
-            permission=data["permission"],
-            origin=data.get("origin"),
-        )
+            if not self._check_grant_subject(identity, new_grant):
```

Comment:
```suggestion
            if not self._validate_grant_subject(identity, new_grant):
```

Reply:
shouldn't this check be inside the `parent.access.grants.create` method before even creating the subject?

Reply:
We can enrich the create method with more validation rules, as for example the subject check. My point is, why to allow a grant creation if afterwards might result to a validation error.